src/layout/mod.rs

```diff
@@ -719,6 +784,34 @@ impl<W: LayoutElement> Layout<W> {
         None
     }
 
+    pub fn find_workspace_by_name(&self, workspace_name: &str) -> Option<(usize, &Workspace<W>)> {
+        if let MonitorSet::Normal { ref monitors, .. } = &self.monitor_set {
+            for mon in monitors {
+                if let Some((index, workspace)) = mon
+                    .workspaces
+                    .iter()
+                    .enumerate()
+                    .find(|(_, w)| w.name == Some(workspace_name.to_string()))
```

Comment:
This is how you avoid allocations in these:
```suggestion
                    .find(|(_, w)| w.name.as_deref() == Some(workspace_name))
```